docs/advanced.md

```diff
@@ -78,11 +78,11 @@ write a predicate function that returns `AssertionResult` instead of `bool`. For
 example, if you define `IsEven()` as:
 
 ```c++
-testing::AssertionResult IsEven(int n) {
+::testing::AssertionResult IsEven(int n) {
```

Comment:
From https://abseil.io/tips/119:
> 1) Never declare namespace aliases or convenience using-declarations at namespace scope in header files, only in `.cc` files.
> 2) Declare namespace aliases and using-declarations inside the innermost namespace, whether named or anonymous. (Do not add an anonymous namespace just for this purpose.)
> 3) When declaring namespace aliases and using-declarations, use fully qualified names (with leading `::`) unless you are referring to a name inside the current namespace.
> 4) For other uses of names, avoid fully qualifying when reasonable

This would be (4), since it's presumably outside `namespace testing`, but neither a namespace alias nor using declaration.
This should apply for the vast majority of the PR that doesn't start with `typedef` or `using`.

Reply:
Thank you for the review!

As you said, for (4), it should be referred to https://abseil.io/tips/130.
And I found that 130 has the followings

> ...
> 
> The current style guide suggests the last option, but allows for the old style (namespaces match package names) if necessary. This is largely because the Google didn’t want to cause too much anxiety or trigger anyone re-namespacing things.
> ...
> 
> Obviously this isn't too bad with std - nobody should ever be adding a nested namespace std - but what about more common namespaces? How about things like testing?

I interpreted it as `::` can be added if a namespace is not guaranteed to be completely unique. And so, I thought `::` should be added if `testing` is used outside its namespace since some projects might already have `testing` in their nested namespaces.